quickstarts/microsoft.compute/vmss-with-public-ip-prefix/main.bicep

```diff
@@ -0,0 +1,231 @@
+  name: virtualNetworkName
+  location: location
+  properties: {
+    addressSpace: {
+      addressPrefixes: [
+        addressPrefix
+      ]
+    }
+    subnets: [
+      {
+        name: subnetName
+        properties: {
+          addressPrefix: subnetPrefix
+        }
+      }
+    ]
+  }
+}
+
+resource publicIPPrefix 'Microsoft.Network/publicIPPrefixes@2021-02-01' = {
+  name: publicIPPrefixName
+  location: location
+  sku: {
+    name: 'Standard'
+    tier: 'Regional'
+  }
+  properties: {
+    prefixLength: publicIPPrefixLength
+    publicIPAddressVersion: 'IPv4'
+  }
+}
+
+resource publicIPAddress 'Microsoft.Network/publicIPAddresses@2021-02-01' = {
+  name: publicIPAddressName
+  location: location
+  sku: {
+    name: 'Standard'
+    tier: 'Regional'
+  }
+  properties: {
+    publicIPAllocationMethod: 'Static'
+    dnsSettings: {
+      domainNameLabel: dnsName
+    }
+  }
+}
+
+resource loadBalancer 'Microsoft.Network/loadBalancers@2020-05-01' = {
+  name: loadBalancerName
+  location: location
+  sku: {
+    name: 'Standard'
+  }
+  properties: {
+    frontendIPConfigurations: [
+      {
+        name: 'LoadBalancerFrontEnd'
+        properties: {
+          publicIPAddress: {
+            id: publicIPAddress.id
+          }
+        }
+      }
+    ]
+    backendAddressPools: [
+      {
+        name: bePoolName
+      }
+    ]
+    inboundNatPools: [
+      {
+        name: natPoolName
+        properties: {
+          frontendIPConfiguration: {
+            id: resourceId('Microsoft.Network/loadBalancers/frontendIPConfigurations', loadBalancerName, 'loadBalancerFrontEnd')
+          }
+          protocol: 'Tcp'
+          frontendPortRangeStart: natStartPort
+          frontendPortRangeEnd: natEndPort
+          backendPort: natBackendPort
+        }
+      }
+    ]
+  }
+}
+
+
+resource vmss 'Microsoft.Compute/virtualMachineScaleSets@2020-06-01' = {
+  name: vmssName
+  location: location
+  sku: {
+    name: vmSku
+    tier: 'Standard'
+    capacity: instanceCount
+  }
+  properties: {
+    overprovision: false
+    upgradePolicy: {
+      mode: 'Manual'
+    }
+    virtualMachineProfile: {
+      storageProfile: {
+        osDisk: {
+          caching: 'ReadOnly'
+          createOption: 'FromImage'
+        }
+        imageReference: imageReference
+      }
+      osProfile: {
+        computerNamePrefix: vmssName
+        adminUsername: adminUsername
+        adminPassword: adminPasswordOrKey
+        linuxConfiguration: ((authenticationType == 'password') ? json('null') : linuxConfiguration)
+      }
+      networkProfile: {
+        networkInterfaceConfigurations: [
+          {
+            name: nicName
+            properties: {
+              primary: true
+              ipConfigurations: [
+                {
+                  name: ipConfigName
+                  properties: {
+                    subnet: {
+                      id: '${virtualNetwork.id}/subnets/${subnetName}'
```

Comment:
use the resourceId() function

Reply:
Hi @bmoore-msft according to the [Bicep Best Practice page](https://docs.microsoft.com/en-us/azure/azure-resource-manager/bicep/best-practices), usage of resourceId and reference functions should be avoided to ensure bicep can generate the dependencies. Are you sure we want to use it ? 
> When possible, avoid using the reference and resourceId functions in your Bicep file. You can access any resource in Bicep by using the symbolic name.

Reply:
@bmoore-msft any opinion on previous message ?